pkg/local_workflows/data_transformation_workflow.go

```diff
@@ -108,63 +108,353 @@ func dataTransformationEntryPoint(invocationCtx workflow.InvocationContext, inpu
+		ignored_email = *suppression.Properties.IgnoredBy.Email
+	}
+	return &local_models.TypesSuppression{
+		Details: &local_models.TypesSuppressionDetails{
+			Category:   string(suppression.Properties.Category),
+			Expiration: expiration,
+			IgnoredOn:  suppression.Properties.IgnoredOn,
+			IgnoredBy: local_models.TypesUser{
+				Name:  suppression.Properties.IgnoredBy.Name,
+				Email: ignored_email,
+			},
+		},
+		Justification: &suppression.Justification,
+		Kind:          "ignored",
+	}
+}
 
-	cliOutput, applyError := api2cliTransformer.ApplyValue(apiOutput)
-	if applyError != nil {
-		return localFinding, applyError
+func mapFingerprints(res sarif.Result) []local_models.Fingerprint {
+	var fingerprints []local_models.Fingerprint
+	if res.Fingerprints.Identity != "" {
+		var fp local_models.Fingerprint
+		rawIdentity := []byte(`{"scheme":"` + string(local_models.Identity) + `","value":"` + res.Fingerprints.Identity + `"}`)
+		if err := json.Unmarshal(rawIdentity, &fp); err != nil {
+			log.Warn().Msg("Failed to unmarshal identity fingerprint")
+		} else {
+			fingerprints = append(fingerprints, fp)
+		}
 	}
+	if res.Fingerprints.Num0 != "" {
+		var fp local_models.Fingerprint
+		rawNum0 := []byte(`{"scheme":"` + string(local_models.CodeSastV0) + `","value":"` + res.Fingerprints.Num0 + `"}`)
+		if err := json.Unmarshal(rawNum0, &fp); err == nil {
+			fingerprints = append(fingerprints, fp)
+		}
+	}
+	if res.Fingerprints.Num1 != "" {
+		var fp local_models.Fingerprint
+		rawNum1 := []byte(`{"scheme":"` + string(local_models.CodeSastV1) + `","value":"` + res.Fingerprints.Num1 + `"}`)
+		if err := json.Unmarshal(rawNum1, &fp); err == nil {
+			fingerprints = append(fingerprints, fp)
+		}
+	}
+	return fingerprints
+}
 
-	// Gate with validation before encoding?
-	encodeErr := cliOutput.Decode(&localFinding)
+func mapCodeFlows(res sarif.Result) []local_models.TypesCodeFlow {
+	var codeFlows []local_models.TypesCodeFlow
+	for _, cf := range res.CodeFlows {
+		var codeFlow local_models.TypesCodeFlow
+		for _, tf := range cf.ThreadFlows {
+			var threadFlow local_models.TypesThreadFlow
+			for _, loc := range tf.Locations {
+				threadFlow.Locations = append(threadFlow.Locations, local_models.IoSnykReactiveFindingSourceLocation{
+					Filepath:            loc.Location.PhysicalLocation.ArtifactLocation.URI,
+					OriginalStartLine:   loc.Location.PhysicalLocation.Region.StartLine,
+					OriginalEndLine:     loc.Location.PhysicalLocation.Region.EndLine,
+					OriginalStartColumn: loc.Location.PhysicalLocation.Region.StartColumn,
+					OriginalEndColumn:   loc.Location.PhysicalLocation.Region.EndColumn,
+				})
+			}
+			codeFlow.ThreadFlows = append(codeFlow.ThreadFlows, threadFlow)
+		}
+		codeFlows = append(codeFlows, codeFlow)
+	}
+	return codeFlows
+}
 
-	if encodeErr != nil {
-		return localFinding, fmt.Errorf("failed to convert to type: %w", encodeErr)
+func mapRules(sarifDoc sarif.SarifDocument) []local_models.TypesRules {
```

Comment:
Suggestion: Use the sarif rules as an input parameter, this way, `mapRules` accepts a sarif rule and outputs a findingsmodel rule. It basically moves the decision of just using a single sarif rule one level up. See [IOSP](https://clean-code-developer.de/en/the-straight/red-degree/#elementor-toc__heading-anchor-5) for a reference.